src/internal/m365/collection/drive/collections_tree.go

```diff
@@ -0,0 +1,325 @@
+package drive
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/prefixmatcher"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/data"
+	bupMD "github.com/alcionai/corso/src/pkg/backup/metadata"
+	"github.com/alcionai/corso/src/pkg/count"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/graph"
+	"github.com/alcionai/corso/src/pkg/services/m365/api/pagers"
+)
+
+// this file is used to separate the collections handling between the previous
+// (list-based) design, and the in-progress (tree-based) redesign.
+// see: https://github.com/alcionai/corso/issues/4688
+
+func (c *Collections) getTree(
+	ctx context.Context,
+	prevMetadata []data.RestoreCollection,
+	ssmb *prefixmatcher.StringSetMatchBuilder,
+	errs *fault.Bus,
+) ([]data.BackupCollection, bool, error) {
+	ctx = clues.AddTraceName(ctx, "GetTree")
```

Comment:
should we prefix this with the service/data type?

Reply:
the trace naturally composes a branch of values within a process tree.  So, ideally, a parent container would have named (or more likely, auto-traced) the service at a higher point in the process.